crates/symbolicator-service/src/services/download/s3.rs

```diff
@@ -88,56 +90,62 @@ impl S3Downloader {
         s3_client_capacity: u64,
     ) -> Self {
         Self {
-            http_client: Arc::new(rusoto_core::HttpClient::new().unwrap()),
             client_cache: ClientCache::new(s3_client_capacity),
             connect_timeout,
             streaming_timeout,
         }
     }
 
-    fn get_s3_client(&self, key: &Arc<S3SourceKey>) -> Arc<rusoto_s3::S3Client> {
+    async fn get_s3_client(&self, key: &Arc<S3SourceKey>) -> Arc<Client> {
         if self.client_cache.contains_key(key) {
             metric!(counter("source.s3.client.cached") += 1);
         }
 
         self.client_cache.get_with_by_ref(key, || {
-            metric!(counter("source.s3.client.create") += 1);
-
-            let region = key.region.clone();
-            tracing::debug!(
-                "Using AWS credentials provider: {:?}",
-                key.aws_credentials_provider
-            );
-            Arc::new(match key.aws_credentials_provider {
-                AwsCredentialsProvider::Container => {
-                    let container_provider = rusoto_credential::ContainerProvider::new();
-                    let provider =
-                        match rusoto_credential::AutoRefreshingProvider::new(container_provider) {
-                            Ok(provider) => provider,
-                            Err(err) => panic!(
-                            "Unable to instantiate rusoto_credential::AutoRefreshingProvider: {:?}",
-                            err
-                        ),
-                        };
-                    self.create_s3_client(provider, region)
-                }
-                AwsCredentialsProvider::Static => {
-                    let provider = rusoto_credential::StaticProvider::new_minimal(
-                        key.access_key.clone(),
-                        key.secret_key.clone(),
-                    );
-                    self.create_s3_client(provider, region)
-                }
+            tokio::runtime::Handle::current().block_on(async {
+                metric!(counter("source.s3.client.create") += 1);
+
+                let region = key.region.clone();
+                tracing::debug!(
+                    "Using AWS credentials provider: {:?}",
+                    key.aws_credentials_provider
+                );
+                Arc::new(match key.aws_credentials_provider {
+                    AwsCredentialsProvider::Container => {
+                        let provider = LazyCachingCredentialsProvider::builder()
+                            .load(provide_credentials_fn(|| async {
+                                aws_config::ecs::EcsCredentialsProvider::builder()
+                                    .build()
+                                    .credentials()
+                                    .await
+                            }))
```

Comment:
```suggestion
                            .load(aws_config::ecs::EcsCredentialsProvider::builder().build())
```

`aws_config::ecs::EcsCredentialsProvider` implements `ProvideCredentials` directly:
https://docs.rs/aws-config/latest/aws_config/ecs/struct.EcsCredentialsProvider.html#impl-ProvideCredentials-for-EcsCredentialsProvider

Reply:
Nice! Fixed, thanks.